ee/localserver/zta_auth_middleware_test.go

```diff
@@ -0,0 +1,336 @@
+package localserver
+
+import (
+	"crypto/ecdsa"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/kolide/launcher/pkg/log/multislogger"
+	"github.com/lestrrat-go/jwx/jwk"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/nacl/box"
+)
+
+func Test_ZtaAuthMiddleware(t *testing.T) {
+	rootTrustedEcKey := mustGenEcdsaKey(t)
+
+	ztaMiddleware := &ztaAuthMiddleware{
+		counterPartyKeys: map[string]*ecdsa.PublicKey{
+			"for_funzies": mustGenEcdsaKey(t).Public().(*ecdsa.PublicKey),
+			"0":           rootTrustedEcKey.Public().(*ecdsa.PublicKey), // this is the trusted root
+		},
+		slogger: multislogger.NewNopLogger(),
+	}
+
+	returnData := []byte("Congrats!, you got the data back!")
+
+	handler := ztaMiddleware.Wrap(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write(returnData)
+		w.WriteHeader(http.StatusOK)
+	}))
+
+	t.Run("handles missing box param", func(t *testing.T) {
+		t.Parallel()
+		rr := httptest.NewRecorder()
+		handler.ServeHTTP(rr, httptest.NewRequest(http.MethodGet, "/", nil))
+		require.Equal(t, http.StatusBadRequest, rr.Code,
+			"should return bad request when box param is missing",
+		)
+	})
+
+	t.Run("handles bad b64", func(t *testing.T) {
+		t.Parallel()
+		rr := httptest.NewRecorder()
+		handler.ServeHTTP(rr, httptest.NewRequest(http.MethodGet, "/?payload=badb64", nil))
+		require.Equal(t, http.StatusBadRequest, rr.Code,
+			"should return bad request when box param is not valid b64",
+		)
+	})
+
+	t.Run("handles chain unmarshall failure", func(t *testing.T) {
+		t.Parallel()
+		rr := httptest.NewRecorder()
+		encoded := base64.URLEncoding.EncodeToString([]byte("badchain"))
+		handler.ServeHTTP(rr, httptest.NewRequest(http.MethodGet, fmt.Sprintf("/?payload=%s", encoded), nil))
+		require.Equal(t, http.StatusBadRequest, rr.Code,
+			"should return bad request when chain cannot be unmarshalled",
+		)
+	})
```

Comment:
Maybe worth adding a similar test confirming we return a `http.StatusBadRequest` for an empty chain? i.e. I think `encoded := base64.URLEncoding.EncodeToString([]byte("[]"))`.

I see there's a test below in `Test_ValidateCertChain` that covers empty chain, so maybe not needed?

Reply:
I think it's a legit test, easy to add, done